src/python/pants/engine/internals/graph_test.py

```diff
@@ -1006,6 +1012,92 @@ def test_generate_overrides_unused(generated_targets_rule_runner: RuleRunner) ->
+                Address("demo", relative_file_path="f1.ext", parameters={"tags": "t2"}),
+                residence_dir="demo",
+            ),
+        },
+    )
+
+
+def test_parametrize_multi(generated_targets_rule_runner: RuleRunner) -> None:
+    assert_generated(
+        generated_targets_rule_runner,
+        Address("demo"),
+        "generator(tags=parametrize(t1=['t1'], t2=['t2']), resolve=parametrize('a', 'b'), sources=['f1.ext'])",
+        ["f1.ext"],
+        {
+            MockGeneratedTarget(
+                {SingleSourceField.alias: "f1.ext", Tags.alias: ["t1"], ResolveField.alias: "a"},
+                Address(
+                    "demo", relative_file_path="f1.ext", parameters={"tags": "t1", "resolve": "a"}
+                ),
+                residence_dir="demo",
+            ),
+            MockGeneratedTarget(
+                {SingleSourceField.alias: "f1.ext", Tags.alias: ["t2"], ResolveField.alias: "a"},
+                Address(
+                    "demo", relative_file_path="f1.ext", parameters={"tags": "t2", "resolve": "a"}
+                ),
+                residence_dir="demo",
+            ),
+            MockGeneratedTarget(
+                {SingleSourceField.alias: "f1.ext", Tags.alias: ["t1"], ResolveField.alias: "b"},
+                Address(
+                    "demo", relative_file_path="f1.ext", parameters={"tags": "t1", "resolve": "b"}
+                ),
+                residence_dir="demo",
+            ),
+            MockGeneratedTarget(
+                {SingleSourceField.alias: "f1.ext", Tags.alias: ["t2"], ResolveField.alias: "b"},
+                Address(
+                    "demo", relative_file_path="f1.ext", parameters={"tags": "t2", "resolve": "b"}
+                ),
+                residence_dir="demo",
+            ),
+        },
+    )
+
+
+def test_parametrize_overrides(generated_targets_rule_runner: RuleRunner) -> None:
+    assert_generated(
+        generated_targets_rule_runner,
+        Address("demo"),
+        "generator(overrides={'f1.ext': {'resolve': parametrize('a', 'b')}}, resolve='c', sources=['*.ext'])",
+        ["f1.ext", "f2.ext"],
+        {
+            MockGeneratedTarget(
+                {SingleSourceField.alias: "f1.ext", ResolveField.alias: "a"},
+                Address("demo", relative_file_path="f1.ext", parameters={"resolve": "a"}),
+                residence_dir="demo",
+            ),
+            MockGeneratedTarget(
+                {SingleSourceField.alias: "f1.ext", ResolveField.alias: "b"},
+                Address("demo", relative_file_path="f1.ext", parameters={"resolve": "b"}),
+                residence_dir="demo",
+            ),
+            MockGeneratedTarget(
+                {SingleSourceField.alias: "f2.ext", ResolveField.alias: "c"},
+                Address("demo", relative_file_path="f2.ext"),
+                residence_dir="demo",
+            ),
+        },
+    )
+
+
```

Comment:
Could you please add some tests of parametrizing an atom target like `python_source`?